job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -272,6 +292,50 @@ private void copyLogs() {
             }
         }
     }
+
+    private String getLogPrefix(FlowNode node) {
+        BodyInvocationAction bodyInvocationAction = node.getAction(BodyInvocationAction.class);
+
+        if (bodyInvocationAction != null) {
```

Comment:
@jglick Using `BlockEndNode` is better but still not quite right. See the `Running: Execute sub-workflows in parallel : Body : End` logs think they have a `ThreadNameAction` parent. Will investigate more later. Thanks Jesse.

```
Started
Running: Allocate node : Start
Running on master in /var/folders/nw/tdwkgqsx1bj91gbwybt4d8540000gn/T/junit6508583420884267681/junit4964712954863688158/workspace/demo
Running: Allocate node : Body : Start
Running: Execute sub-workflows in parallel : Start
[a] Running: Parallel branch: a
[b] Running: Parallel branch: b
[a] Running: Print Message
[a] echo a
[a] Running: Print Message
[a] echo a
[a] Running: Execute sub-workflows in parallel : Body : End
[b] Running: Print Message
[b] echo b
[b] Running: Print Message
[b] echo b
[b] Running: Execute sub-workflows in parallel : Body : End
Running: Execute sub-workflows in parallel : End
Running: Allocate node : Body : End
Running: Allocate node : End
Running: End of Workflow
Finished: SUCCESS
```